include/wil/chrono.h

```diff
@@ -0,0 +1,370 @@
+//*********************************************************
+//
+//    Copyright (c) Microsoft. All rights reserved.
+//    This code is licensed under the MIT License.
+//    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF
+//    ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED
+//    TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+//    PARTICULAR PURPOSE AND NONINFRINGEMENT.
+//
+//*********************************************************
+#ifndef __WIL_CHRONO_INCLUDED
+#define __WIL_CHRONO_INCLUDED
+
+#include "common.h"
+#include "result.h"
+
+#ifdef WIL_KERNEL_MODE
+#error This header is not supported in kernel-mode.
+#endif
+
+#if __WI_CPLUSPLUS < 201103L
+#error This header requires C++11 or later
+#endif
+
+#include <chrono>
+
+#include <minwindef.h> // FILETIME, etc.
+#include <sysinfoapi.h> // GetTickCount[64], GetSystemTime[Precise]AsFileTime
+#include <realtimeapiset.h> // Query[Unbiased]InterruptTime[Precise]
+
+namespace wil
+{
+#pragma region std::chrono wrappers for GetTickCount[64]
+    namespace details
+    {
+        template<typename rep_t, typename get_tick_count_fn_t, get_tick_count_fn_t get_tick_count_fn, typename base_clock_t>
+        struct tick_count_clock_impl
+        {
+            using rep = rep_t;
+            using period = std::milli;
+            using duration = std::chrono::duration<rep, period>;
+            using time_point = std::chrono::time_point<base_clock_t, duration>;
+            static constexpr bool const is_steady = true;
+
+            WI_NODISCARD static time_point now() WI_NOEXCEPT
+            {
+                return time_point{duration{ get_tick_count_fn() }};
+            }
+        };
+    }
+
+    struct tick_count_clock : details::tick_count_clock_impl<DWORD, decltype(&::GetTickCount), ::GetTickCount, tick_count_clock> {};
+#if _WIN32_WINNT >= 0x0600
+    struct tick_count64_clock : details::tick_count_clock_impl<ULONGLONG, decltype(&::GetTickCount64), ::GetTickCount64, tick_count_clock> {};
+#endif // _WIN32_WINNT >= 0x0600
+#pragma endregion
+
+    namespace details
+    {
+        constexpr DWORD64 filetime_to_int(const FILETIME& ft) WI_NOEXCEPT

```

Comment:
This functionality is already present in `win32_helpers.h`

Reply:
The author has noted this in their initial pull request comment, saying that they feel including win32_helpers.h (and the bulk of resource.h) just for a single line was not worth it.

Reply:
The win32_helpers.h routines are also non-constexpr, and to be made constexpr they'd have to be rewritten not to use reinterpret_cast. I'd rather not touch an unrelated header

Reply:
They should be totally rewritten to just bit shift IMO (and made constexpr while at it), I know quite a few people who would be ready to complain that the reinterpret_cast is UB, been flamed by them for trivial a union type pun before :P